services/provider/server/server.go

```diff
@@ -681,12 +707,28 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				"clusterID":          getSubVolumeGroupClusterID(subVolumeGroup),
 				"subvolumegroupname": subVolumeGroup.Name,
 				"fsName":             subVolumeGroup.Spec.FilesystemName,
-				"pool":               subVolumeGroup.GetLabels()[v1alpha1.CephFileSystemDataPoolLabel],
+				"pool":               subVolumeGroup.GetLabels()[ocsv1alpha1.CephFileSystemDataPoolLabel],
 				"csi.storage.k8s.io/provisioner-secret-name":       provisionerSecretName,
 				"csi.storage.k8s.io/node-stage-secret-name":        nodeSecretName,
 				"csi.storage.k8s.io/controller-expand-secret-name": provisionerSecretName,
 			}
 
+			storageClusters := &ocsv1.StorageClusterList{}
+			if err := s.client.List(ctx, storageClusters, client.InNamespace(s.namespace), client.Limit(2)); err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to get storage cluster: %v", err)
+			}
+			if len(storageClusters.Items) != 1 {
+				return nil, status.Errorf(codes.Internal, "expecting one single storagecluster to exist")
+			}
+			var kernelMountOptions map[string]string
+			for _, option := range strings.Split(util.GetCephFSKernelMountOptions(&storageClusters.Items[0]), ",") {
+				if kernelMountOptions == nil {
+					kernelMountOptions = map[string]string{}
+				}
+				parts := strings.Split(option, "=")
```

Comment:
we need if check for parts length and i don't understand if check for kernelMountOptions above, when that can happen?

Reply:
@Madhu-1 Are we assuming that the mount options will be returned malformed?

Reply:
Regarding the check. I have asked to ensure that we will serialize a nil map and not an empty one in case of no mount options. This means that we need to start with a nil map and allocate a map only if we encounter a mount option 

Please see my comment here: https://github.com/red-hat-storage/ocs-operator/pull/2713#discussion_r1713733642

Reply:
it won't be but if the function changed to something else or a bug, this assumption breaks with panic.

Reply:
If the function changes in a way that breaks the format of the KMO then it should panic so the dev would notice and fix it. If the format changes this code should change as well. Returning an error at runtime will not be visible until runtime which is too late